pkg/sql/region_util.go

```diff
@@ -1468,6 +1474,9 @@ func SynthesizeRegionConfig(
 	if err != nil {
 		return regionConfig, err
 	}
+	if !dbDesc.IsMultiRegion() {
+		return regionConfig, errNotMultiRegionDatabase
+	}
```

Comment:
before this change, we'd return an assertion failure error, so this is strictly better